### PR TITLE
Us27 handle users closing

### PR DESCRIPTION
**Pull request by grorge**
Made it possible for users to close down and rejoin the game.
When a user joins they will get the earliest index that is free and gets assigned that playerX
If more than 4 users join it will print in the console and not execute their command.

The delay can sometimes get very slow, might be because of the thread not getting scheduled fast enough.

### CODE REVIEW

File: InputHandler/InputHandler/includes/webConnection.h [hunk 1]
```diff
@@ -172,8 +178,9 @@ class WebConnection
 	// interpets the key and preformes handshake
 	bool checkForKey(SOCKET sock, char* recBuff, int& Res);
 
-	int nrOfPlayers;
-	SOCKET playerSockets[30];
+	//int nrOfPlayers;
+	const int mMaxUserSockets = 30;
+	SOCKET userSockets[30];
```
Comment (NistBox) [inline InputHandler/InputHandler/includes/webConnection.h]: Missing prepended 'm'

File: InputHandler/InputHandler/webConnection.cpp [hunk 2]
```diff
@@ -219,89 +228,49 @@ void WebConnection::playersJoin()
...
+							cout << "-A key was sent in joining phase" << endl;
+						}
+						// if the socket is the listener in the array
+						else
+						{
+							//cout << endl << "___We just got a message from " << this->idUserSocket(sock) << endl;
 
-						char* client_msg = this->reciveMsg(sock, recvbuf, iSendResult);
-						this->msgToClient = client_msg;
+							char* userMsg = reciveMsg(sock, recvbuf, iSendResult);
```
Comment (NistBox) [inline InputHandler/InputHandler/webConnection.cpp]: Stack variable, userMsg -> user_msg

File: InputHandler/InputHandler/webConnection.cpp [hunk 3]
```diff
@@ -346,85 +310,57 @@ void WebConnection::gameLoop()
...
 					}
+					// if the socket is the listener in the array
 					else
 					{
-						wmd.player = idPlayerSocket(sock);
+						//cout << endl << "___We just got a message from " << this->idUserSocket(sock) << endl;
 
-						cout << "-From player " << wmd.player << endl
-							<< userMsg << endl;
+						char* userMsg = reciveMsg(sock, recvbuf, iSendResult);
```
Comment (NistBox) [inline InputHandler/InputHandler/webConnection.cpp]: Stack variable, userMsg -> user_msg

File: InputHandler/InputHandler/webConnection.cpp [hunk 4]
```diff
@@ -664,12 +612,97 @@ void WebConnection::sendMsg(SOCKET sock, char* client_msg, int& Res)
...
+			return true;
+		}
+	}
+	
+	return false;
+}
+
+bool WebConnection::addUserSocket(SOCKET sock)
+{
+	int firstEmpty = 0;
```
Comment (NistBox) [inline InputHandler/InputHandler/webConnection.cpp]: Stack variable, firstEmpty-> first_empty

**Comment by grorge** [comment]
try nr 2 to fix

**Pull request merged**